sjb/hack/determine_install_upgrade_version_test.py

```diff
@@ -158,7 +158,7 @@ def test_sort_packages_with_exceptional_origin_pkg(self):
 		test_pkgs_obj = TestPackage.create_test_packages(test_pkgs)
 		properly_sorted_pkgs_obj = TestPackage.create_test_packages(properly_sorted_pkgs)
 		sorted_test_pkgs_obj = sort_pkgs(test_pkgs_obj)
-		self.assertEqual(sorted_test_pkgs_obj, properly_sorted_pkgs_obj)
+		self.assertTrue(sorted_test_pkgs_obj[0], properly_sorted_pkgs_obj[0])
```

Comment:
That's not true. `unittest.assertEqual` will (for lists) fall back to [`unitttest.assertListEqual`](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertListEqual) which [uses the `==` operator](http://asynctest.readthedocs.io/en/latest/_modules/unittest/case.html) and not the `is` operator:

```py
    def assertListEqual(self, list1, list2, msg=None):
        """A list-specific equality assertion.

        Args:
            list1: The first list to compare.
            list2: The second list to compare.
            msg: Optional message to use on failure instead of a list of
                    differences.

        """
        self.assertSequenceEqual(list1, list2, msg, seq_type=list)

    def assertSequenceEqual(self, seq1, seq2, msg=None, seq_type=None):
        """An equality assertion for ordered sequences (like lists and tuples).

        For the purposes of this function, a valid ordered sequence type is one
        which can be indexed, has a length, and has an equality operator.

        Args:
            seq1: The first sequence to compare.
            seq2: The second sequence to compare.
            seq_type: The expected datatype of the sequences, or None if no
                    datatype should be enforced.
            msg: Optional message to use on failure instead of a list of
                    differences.
        """
        if seq_type is not None:
            seq_type_name = seq_type.__name__
            if not isinstance(seq1, seq_type):
                raise self.failureException('First sequence is not a %s: %s'
                                        % (seq_type_name, safe_repr(seq1)))
            if not isinstance(seq2, seq_type):
                raise self.failureException('Second sequence is not a %s: %s'
                                        % (seq_type_name, safe_repr(seq2)))
        else:
            seq_type_name = "sequence"

        differing = None
        try:
            len1 = len(seq1)
        except (TypeError, NotImplementedError):
            differing = 'First %s has no length.    Non-sequence?' % (
                    seq_type_name)

        if differing is None:
            try:
                len2 = len(seq2)
            except (TypeError, NotImplementedError):
                differing = 'Second %s has no length.    Non-sequence?' % (
                        seq_type_name)

        if differing is None:
            if seq1 == seq2:
                return
            // more code if they are not equal 
```

In Python, `==` checks for object equality while `is` will do [object identity](https://docs.python.org/2/library/stdtypes.html#comparisons) as well.